src/OpenTelemetry.Abstractions/Trace/ITracer.cs

```diff
@@ -50,107 +51,71 @@ public interface ITracer
         // TODO: add sampling hints
 
         /// <summary>
-        /// Creates root span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="linksGetter">Func to retrieve the Links collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartRootSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, Func<IEnumerable<Link>> linksGetter);
-
-        /// <summary>
-        /// Creates root span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="links">collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartRootSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, IEnumerable<Link> links);
-
-        /// <summary>
-        /// Creates span. If there is active current span, it becomes a parent for returned span.
+        /// Starts root span.
         /// </summary>
         /// <param name="operationName">Span name.</param>
         /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="linksGetter">Func to retrieve the Links collection.</param>
+        /// <param name="options">Advanced span creation options.</param>
         /// <returns>Span instance.</returns>
-        ISpan StartSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, Func<IEnumerable<Link>> linksGetter);
+        ISpan StartRootSpan(string operationName, SpanKind kind, SpanCreationOptions options);
 
         /// <summary>
-        /// Creates span. If there is active current span, it becomes a parent for returned span.
+        /// Starts activate root span.
         /// </summary>
         /// <param name="operationName">Span name.</param>
         /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="links">Links collection.</param>
+        /// <param name="options">Advanced span creation options.</param>
         /// <returns>Span instance.</returns>
-        ISpan StartSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, IEnumerable<Link> links);
+        IDisposable StartActiveRootSpan(string operationName, SpanKind kind, SpanCreationOptions options);
```

Comment:
What's the motivation to have this method?

Reply:
Would it be the same to add some extension method instead of making it a part of an API?

Reply:
> What's the motivation to have this method?

Motivation is explained in the PR description: friendy and short API for the most used scenarios. Is your question about Root span in particular? I can see that `StartRoot` is rare, so we don't have to have it. 

> Would it be the same to add some extension method instead of making it a part of an API?
I hoped for it, but it returns Disposable, not ISpan. Moving it to extensions could be possible if we make ISpan disposable, but I don't see a good reason for that.

Reply:
Extension method can do two operations - Start span normally and than call WithSpan and return the result of this call. I'm not sure why it's not possible. Can you comment?

Reply:
I want to optimize and not allocate another IDisposable. Also, I want to overcome requirement/confusion that scope lifetime is different than span - In the majority of cases, you need the same lifetime.